### PR TITLE
Numpy compatible linspace

### PR DESCRIPTION
**Pull request by stu1130**
## Description ##
Numpy compatible linspace

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
@haojin2 @reminisce

### CODE REVIEW

File: python/mxnet/_numpy_op_doc.py [hunk 1]
```diff
@@ -86,3 +86,50 @@ def _np_zeros_like(a):
         Array of zeros with the same shape and type as `a`.
     """
     pass
+
+
+def _npi_linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None, axis=0):
```
Comment (reminisce) [inline python/mxnet/_numpy_op_doc.py]: This is not needed. We only need to put ops with prefixes `_np_` and `_npx_` here.

Comment (stu1130) [reply python/mxnet/_numpy_op_doc.py]: done!

File: tests/python/unittest/test_numpy_op.py [hunk 2]
```diff
@@ -554,6 +554,65 @@ def hybrid_forward(self, F, x):
...
+                        net = TestLinspace(*config, endpoint=endpoint, dtype=dtype)
+                        np_out = _np.linspace(*config, endpoint=endpoint, dtype=dtype)
+                    else:
+                        net = TestLinspace(config, endpoint=endpoint, dtype=dtype)
+                        np_out = _np.linspace(config, endpoint=endpoint, dtype=dtype)
+                    if hybridize:
+                        net.hybridize()
+                    mx_out = net(x)
+                    assert_almost_equal(mx_out.asnumpy(), np_out, atol=1e-3, rtol=1e-5)
+
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_op.py]: nit: one more blank line here

File: tests/python/unittest/test_numpy_op.py [hunk 3]
```diff
@@ -554,6 +554,65 @@ def hybrid_forward(self, F, x):
                 assert same(mx_out.asnumpy(), np_out)
 
 
+@with_seed()
+@npx.use_np_shape
+def test_np_linspace():
+    configs = [
+        (0.0, 1.0, 10),
+        (-2, 4, 45),
+        (5.234324, 8.98324, 324)
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_op.py]: Does these 3 cases provide full coverage of all cases? Maybe we can take a look at numpy's test for linspace as a reference?

Comment (stu1130) [reply tests/python/unittest/test_numpy_op.py]: checked the numpy unit test, and added more here

File: tests/python/unittest/test_numpy_op.py (hunk 2 above)
Comment (stu1130) [reply tests/python/unittest/test_numpy_op.py]: done

Comment (stu1130) [reply tests/python/unittest/test_numpy_op.py]: done

**Pull request merged**